File: server/src/SwipeJobs.Domain/Entities/UserProfile.cs
using SwipeJobs.Domain.Common;

namespace SwipeJobs.Domain.Entities;

public class UserProfile : BaseEntity
{
    public Guid? UserId { get; set; }
    public User? User { get; set; }

    public string? ExternalUserId { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public string? Bio { get; set; }
    public string? ResumeUrl { get; set; }
    public string? Location { get; set; }
    public bool IsProfileComplete { get; set; }

    public ICollection<Education> Educations { get; set; } = new List<Education>();
    public ICollection<Skill> Skills { get; set; } = new List<Skill>();
    public ICollection<Experience> Experiences { get; set; } = new List<Experience>();
    public ICollection<Application> Applications { get; set; } = new List<Application>();
    public ICollection<SavedJob> SavedJobs { get; set; } = new List<SavedJob>();
    public ICollection<UserActivity> Activities { get; set; } = new List<UserActivity>();
    public UserInterestProfile? InterestProfile { get; set; }
    public ICollection<CompanyFollow> CompanyFollows { get; set; } = new List<CompanyFollow>();
    public ICollection<Notification> Notifications { get; set; } = new List<Notification>();
}

File: client/src/models/userProfile.ts
export interface Education {
  id?: string;
  institution: string;
  degree: string;
  fieldOfStudy?: string;
  startDate?: string;
  endDate?: string;
  isCurrent: boolean;
}

export interface Skill {
  id?: string;
  name: string;
  level?: string;
}

export interface Experience {
  id?: string;
  company: string;
  title: string;
  description?: string;
  startDate?: string;
  endDate?: string;
  isCurrent: boolean;
}

export interface UserProfile {
  id: string;
  userId?: string;
  externalUserId?: string;
  firstName: string;
  lastName: string;
  email: string;
  phone?: string;
  bio?: string;
  resumeUrl?: string;
  location?: string;
  isProfileComplete: boolean;
  educations: Education[];
  skills: Skill[];
  experiences: Experience[];
  createdAt: string;
  updatedAt?: string;
}

export interface CreateUserProfileRequest {
  externalUserId: string;
  firstName: string;
  lastName: string;
  email: string;
  phone?: string;
  bio?: string;
  resumeUrl?: string;
  location?: string;
  educations?: Education[];
  skills?: Skill[];
  experiences?: Experience[];
}

export interface UpdateUserProfileRequest {
  firstName: string;
  lastName: string;
  email: string;
  phone?: string;
  bio?: string;
  resumeUrl?: string;
  location?: string;
  educations?: Education[];
  skills?: Skill[];
  experiences?: Experience[];
}

export interface ProfileCompleteness {
  isComplete: boolean;
  missingFields: string[];
}
